atc/worker/gardenruntime/gardenruntimetest/baggageclaim.go

```diff
@@ -7,16 +7,20 @@ import (
 	"net/http"
 	"net/http/httptest"
 	"strings"
+	"sync"
 
 	"github.com/concourse/concourse/atc/runtime/runtimetest"
 	"github.com/concourse/concourse/worker/baggageclaim"
 )
 
 type Baggageclaim struct {
 	Volumes []*Volume
+	Mutex   sync.Mutex
```

Comment:
Should `FilteredVolumes()` also be mutex protected? In that case, `RWMutex` might be a better fit. Or maybe we can try https://github.com/cornelk/hashmap. As Volumes are compared by handle, I guess using a hashmap with handle as key might be more performant.

Reply:
Probably can yeah. To be honest since this is test code I just made it thread-safe enough to pass the `-race` tests - the real baggageclaim deals with this with proper locks and things.

Reply:
@andy-paine To clarify, if you mean that eventually this PR will not include changes in this file?

Reply:
The PR tests failed due to race conditions https://ci.concourse-ci.org/teams/contributor/pipelines/pr/jobs/unit/builds/1?vars.number=8047#L61fdb20d:1096 so that's why I introduced this in this PR as otherwise it wouldn't pass checks

Reply:
That's weird. I applied this PR to my local and ran unit test, but didn't hit the race error at all. Can you try remove the change and see if unit test still hit the error?